### PR TITLE
Enhance Documentation

### PR DESCRIPTION
**Pull request by shubhwip**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
 * Update docs to help Apple Mac M1 Users
 * Update missing information about building react app
 * Update prerequisites docs

Some of the steps from `building` and `running` document were overlapping and causing some confusion thus I propose to have one file as currently as our steps are limited.
There were some steps missing for Apple macOS M1. Included them.

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [x] No need to
- [ ] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: documentation/project/contributing/building.md [hunk 1]
```diff
@@ -1,24 +1,62 @@
...
+
+> Comment out `com.spotify` plugin in `kafka-ui-api` pom file
+
+> Once your build is successful you need to create docker image manually as you'd commented `com.spotify` plugin
+
+> ```docker build --platform linux/amd64 -t provectuslabs/kafka-ui:latest --build-arg JAR_FILE=kafka-ui-api-0.0.1-SNAPSHOT.jar .```
+
+
+- Build `kafka-ui-react-app`
+Follow all the steps [kafka-ui-react-app-build-documentation](../../../kafka-ui-react-app/README.md)
```
Comment (Haarolean) [inline documentation/project/contributing/building.md:18]: uuhm that'd be nice to provide a link like this here but it looks like it's a required step, but the previous one was to build the whole jar. I'd suggest naming this part like "if you need to build the frontend, go here"

**Review by Haarolean — APPROVED** [review]

File: documentation/project/contributing/building.md (hunk 1 above)
Comment (Haarolean) [reply documentation/project/contributing/building.md:18]: @shubhwip can we update this part a bit?:)

Comment (shubhwip) [reply documentation/project/contributing/building.md:18]: Sure Ok :) I remember generate sources was still mandatory step. should i include `pnpm gen:sources` as a necessary step prior to building and running backend ?

File: documentation/project/contributing/building.md [hunk 2]
```diff
@@ -1,24 +1,62 @@
...
+In case you want to build by skipping the tests
+```sh
+./mvnw clean install -Dmaven.test.skip=true -Pprod
 ``` 
-Start the app with Kafka clusters:
+If this step is successful, it should create a docker image named `provectuslabs/kafka-ui` with `latest` tag on your local machine except macOS M1.
+
+#### Step 2 : Run
+##### Using Docker Image
+> **_NOTE:_**  If you are an macOS M1 User then to have smooth experience replace image tag in `./documentation/compose/kafka-ui.yaml` with arm64 compatible images and you can find them [ARM64 Compatible Image Tags](https://hub.docker.com/r/confluentinc/cp-kafka/tags?page=1&name=arm64)
```
[the remark this answers is not in the document]
Comment (shubhwip) [reply documentation/project/contributing/building.md:33]: Ok i will add a compose file in issue #2502. Please assign this to me. So once we have added the docker compose file for arm64 then we can point that file directly to M1 users.

File: documentation/project/contributing/building.md (hunk 1 above)
Comment (Haarolean) [reply documentation/project/contributing/building.md:18]: I'm not quite sure that it's always the case, it's present in both contract and api modules (full search for `pnpm gen:sources`). I dunno why it was a problem in your case.

Comment (shubhwip) [reply documentation/project/contributing/building.md:18]: Yeah true i also saw that. Let me try again with clean project and see if this is a problem still but for now we can update with your suggested changes.

File: documentation/project/contributing/building.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Haarolean) [reply documentation/project/contributing/building.md:33]: @shubhwip btw can we also note that m1/arm users would better install arm jdk?

**Comment by JIBIN-P** [comment]
Hii this PR looks good and answers all the trouble we faced during building the app without docker. Till the time it is merged can anybody just reference it in the documentation or is there a timeline for it to be merged?

**Comment by Haarolean** [comment]
@JIBIN-P merged

**Comment by Haarolean** [comment]
@shubhwip thank you very much
